### ISSUE TITLE
Node.js example

### ISSUE THREAD

**Issue by Siyfion** [metafloor/bwip-js#6]
I was just thinking to myself about the Node.js server example and I realised that the server will be running entirely on one thread (as is the Node.js way). But I assume that the `bwipjs(req, res);` involves a fair amount of processing?

So I feel like this example (& the real server that runs it) could be made potentially much more efficient by utilising the new `cluster` module in Node.js (https://nodejs.org/api/cluster.html).

We could spawn a worker for each core that the server has and use these for the processing...? I'm still looking as to how we'd actually go about doing this, but it could lead to some pretty impressive performance gains, with minimal code addition.

**Comment by Siyfion** [comment]
Looks as though it's as simple as:

```javascript
// Simple HTTP server that renders bar code images using bwip-js.
var cluster = require('cluster');
var http   = require('http');
var numCPUs = require('os').cpus().length;
var bwipjs = require('bwip-js');

if (cluster.isMaster) {
    // Fork workers.
    for (var i = 0; i < numCPUs; i++) {
        cluster.fork();
    }

    cluster.on('exit', function(worker, code, signal) {
        console.log('worker ' + worker.process.pid + ' died');
    });
} else {
    // All the workers will share the HTTP server
    http.createServer(function (req, res) {
        // If the url does not begin /?bcid= then 404.  Otherwise, we end up
        // returning 400 on requests like favicon.ico.
        if (req.url.indexOf('/?bcid=') != 0) {
            res.writeHead(404, { 'Content-Type': 'text/plain' });
            res.end('BWIP-JS: Unknown request format.', 'utf8');
        } else {
            bwipjs(req, res);
        }
    }).listen(3030);
}
```

**Comment by metafloor** [comment]
The 2.0 release provides an example multi-threaded server (using `require("worker_threads")`) available in nodejs 10.5 and beyond.  I have also included this cluster example as well.
